staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go

```diff
@@ -107,7 +107,7 @@ func (t *azureDiskCSITranslator) TranslateInTreeInlineVolumeToCSI(volume *v1.Vol
 		ObjectMeta: metav1.ObjectMeta{
 			// Must be unique per disk as it is used as the unique part of the
 			// staging path
-			Name: fmt.Sprintf("%s-%s", AzureDiskDriverName, azureSource.DiskName),
+			Name: azureSource.DataDiskURI,
```

Comment:
Do you have a link to an issue that describes more in detail the problem and its symptoms? I'm trying to get more context on when the issue could happen.

Reply:
the error is like this
![image](https://user-images.githubusercontent.com/4178417/150462133-324b38f1-2f08-48c4-85a8-118e39837ef5.png)

When there are two volumes with same volumeHandle(volumeid), only one `NodeStageVolume` would be be called on one node if these two volumes are scheduled to the same node, this PR would make volumeid unique in csi translation to solve this issue.

related issue: https://github.com/kubernetes-sigs/azurefile-csi-driver/issues/620

Reply:
So azureSource.DiskName could be the same because the sharename could be specified on StorageClass and that would make the DiskName same?

How is this change related to https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/648/files

Reply:
correct, we should not rely on diskName only, use diskURI is ok since it's unique, and similar to azure file, use pv name as part of volumeit could make volumeid unique

Reply:
I see. So the fix here is basically a fix for a potential issue but we have not encountered a specific bug, right?

Reply:
Does https://github.com/kubernetes/kubernetes/pull/107065 solve this problem?

Reply:
Yes, I think #107065 would fix this issue also. While that PR is only on 1.24, need to fix it in also k8s versions also. Also, there is no harm to make volumeid unique.

Reply:
What is the syntax/format for data disk uri? Want to make sure it will work as a directory name.

Reply:
it's like `/subscriptions/xxx/resourceGroups/mc_andy-aks121canary_andy-aks121canary_eastus2euap/providers/Microsoft.Compute/disks/pvc-a8564491-915f-4613-8efb-d02cce0103aa`

Reply:
@msau42 could you approve this PR? thanks.

Reply:
Thanks for the details!